src/createServer.js

```diff
@@ -1,3 +1,60 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+/* eslint-disable max-len */
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    const url = new URL(`http://${process.env.HOST ?? 'localhost'}${req.url}`);
+
+    const originalText = url.pathname.slice(1);
+    const targetCase = url.searchParams.get('toCase');
+
+    const errors = [];
+
+    if (!originalText) {
+      errors.push({
+        message:
+          'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message:
+          '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else if (
+      !['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'].includes(targetCase)
+    ) {
+      errors.push({
+        message: `This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.`,
+      });
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const { originalCase, convertedText } = convertToCase(
+      originalText,
+      targetCase,
```

Comment:
The parameters for the `convertToCase` function are in the wrong order. According to the task description, the function accepts the case name as the first parameter and the text to convert as the second parameter. Please swap the order of `originalText` and `targetCase` when calling `convertToCase`.